airflow/operators/python.py

```diff
@@ -173,7 +173,9 @@ def __init__(
         self.show_return_value_in_logs = show_return_value_in_logs
 
     def execute(self, context: Context) -> Any:
-        context.update(self.op_kwargs)
+        # The casting below is not `perfect`, but we "pretend" that context is TypedDict via Context.pyi
+        # So no harm done and it will work as expected.
+        context.update(cast(Context, self.op_kwargs))
         context['templates_dict'] = self.templates_dict
```

Comment:
Another possibility is

```suggestion
        context_additions = {"templates_dict": self.templates_dict}
        if self.op_kwargs:
            context_additions.update(self.op_kwargs)
            context = Context(context, **context_additions)
```

It’s probably a good idea to add a function in `airflow.utils.context` for this. Maybe call it `context_merge`.

Reply:
Ah yeah. I like it more :)